functions.R

```diff
@@ -0,0 +1,142 @@
+    geom_line() +
+    geom_hline(yintercept = caaqs, color = "red", lty = "dashed") +
+    annotate(geom = "text", label = "CAAQS", x = 2016, y = caaqs + 5) +
+    labs(
+      x = "Year",
+      y = "Ozone concentration (ppb)",
+      title = paste0(airzone, " Air Zone ", min_year, "-", max_year),
+      colour = "Location"
+    ) +
+    scale_x_continuous(
+      breaks = function(x) floor(pretty(seq(min(x), max(x), by = 1)))
+    ) +
+    scale_y_continuous(limits = c(0, max(ymax_oz + 5, 72))) +
+    theme(plot.title = element_text(hjust = 0.5))
+}
+
+## PM2.5 concentrations bar charts (24-hour concentration and annual mean
+## concentration (fig. 4)
+plot_pm25_by_station <- function(data, caaqs_24h = 28, caaqs_annual = 10) {
+  ## Filter data
+  pm25_24h <- filter(data, metric == "pm2.5_24h")
+  ymax_pm25_24h <- max(pm25_24h$metric_value_ambient, na.rm = TRUE)
+
+  pm25_annual <- filter(data, metric == "pm2.5_annual")
+  ymax_pm25_annual <- max(pm25_annual$metric_value_ambient, na.rm = TRUE)
+
+  ## 24 hour
+  plot_24h <- ggplot(
+    pm25_24h,
+    aes(x = reorder(station_name, metric_value_ambient, sum), y = metric_value_ambient)
+  ) +
+    geom_bar(stat = "identity", aes(fill = instrument_type)) +
+    geom_text(aes(label = metric_value_ambient), nudge_y = 2) +
+    geom_hline(yintercept = caaqs_24h, color = "red", lty = "dashed") +
+    annotate(geom = "text", label = "24h CAAQS", x = 2, y = caaqs_24h + 9) +
+    scale_y_continuous(limits = c(0, max(ymax_pm25_24h + 1, caaqs_24h + 5))) +
+    scale_fill_manual(values = c(FEM = "#4A8CE1", TEOM = "#070C72")) +
+    labs(
+      x = NULL,
+      y = NULL,
+      fill = "Instrument",
+      title = "24-Hr PM2.5"
+    ) +
+    coord_flip() +
+    theme(plot.title = element_text(hjust = 0.5))
+
+  ## Annual
+  plot_annual <- ggplot(
+    pm25_annual,
+    aes(x = reorder(station_name, metric_value_ambient, sum), y = metric_value_ambient)
+  ) +
+    geom_bar(stat = "identity", aes(fill = instrument_type)) +
+    geom_text(aes(label = metric_value_ambient), nudge_y = 0.5) +
+    geom_hline(yintercept = caaqs_annual, color = "red", lty = "dashed") +
+    annotate(geom = "text", label = "Annual\nCAAQS", x = 2, y = caaqs_annual + 2) +
+    scale_y_continuous(limits = c(0, max(ymax_pm25_annual + 1, caaqs_annual + 5))) +
+    scale_fill_manual(values = c(FEM = "#81EDA1", TEOM = "#4A875B")) +
+    labs(
+      x = NULL,
+      y = expression(PM[2.5]~Concentration~"("*mu~g/m^3~")"),
+      fill = "Instrument",
+      title = "Annual PM2.5"
+    ) +
+    coord_flip() +
+    theme(plot.title = element_text(hjust = 0.5))
+
+  plot_24h / plot_annual
+
+}
+
+## Annual trends in mean PM2.5 concentration line chart
+plot_pm25_station_timeseries <- function(data, airzone, caaqs_annual = 10) {
+  ymax_pm25 <- max(data$ann_avg)
+  min_year <- min(data$year, na.rm = TRUE)
+  min_year <- min(data$year, na.rm = TRUE)
+
+  ggplot(annual_pm25, aes(x = year, y = ann_avg, colour = station_name)) +
```

Comment:
I think this needs to use `data` rather than `annual_pm25`:

```suggestion
  ggplot(data, aes(x = year, y = ann_avg, colour = station_name)) +
```

Reply:
Oops yes, thanks!